### PR TITLE
Add renogen

### PR DESCRIPTION
**Pull request by SeanArmstrong**
# Example Output:

# v3.1.0 (2018-02-12)
### Bug Fixes
* `mapToProps` takes precedence over props passed to HOC in `connect` function.

### Other Improvements
* Resolved new ESLint errors from carbon-factory upgrade
* Input Validation: The InputValidation error tooltips now use the Portal Component

### Component Changes
* Modal: The Modal now uses the Portal Component
* Toast: Toast component now uses the Portal Component
* Input: Input has 2 new props. `onChangeDeferred` allows a deferred callback after an onChange event. `deferTimeout` allows you to customise the default: `750`.
* Form

  Form autoDisables after submit when the prop `autoDisable` is set to true. The props `afterFormValidation` and `onSubmit` are passed a `enableForm` callback function which can be used to reactivate the form.

  Example Code
  ```
    <Form
      onSubmit={ this.saveContact }
      autoDisable
    >
      {children}
    </Form>
  ```

  ```
    saveContact = (ev, valid, enableForm) => {
      ...
      Actions.submitForm(...);
      enableForm();
    };
  ```

* Dropdown: Component now uses the Portal Component

https://github.com/Sage/carbon/issues/905

### CODE REVIEW

File: script/generate-release-notes.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+if [ $# -eq 0 ]; then
```
Comment (andrewjtait) [inline script/generate-release-notes.sh]: this version is better

```
#!/bin/bash

PACKAGE_VERSION=$(cat package.json \
  | grep version \
  | head -1 \
  | awk -F: '{ print $2 }' \
  | sed 's/[",]//g')

dot_version=`echo $PACKAGE_VERSION | sed -E "s/-rc.*$//"`
dash_version=`echo ${dot_version} | tr '.' '_'`

RELEASE_NOTES_FILE="release_notes/v${dash_version}.html.md"

renogen --format markdown v${dot_version} >> ${RELEASE_NOTES_FILE}

mkdir change_log/v${dash_version}
mv change_log/next/* change_log/v${dash_version}/

cat ${RELEASE_NOTES_FILE}
```

**Comment by chrisbarber86** [comment]
we should look at adding/running `renogen test` as part of the CI - this would prove that release notes submitted are valid at the point of pull request - rather than trying to fix problems at the end when generating them

File: README.md [hunk 2]
```diff
@@ -58,14 +58,24 @@ To run the example, do the following steps:
...
+
 ## Submitting a Release
 
 * N.B. Releases are created by Sage Carbon Team. We use [semantic versioning](http://semver.org/)
 
 * If releasing a minor version, create a branch from `master`.
 * If releasing a patch version, create a branch from the tag you want to patch. This should be the latest tag apart from exceptional circumstances.
 * Bump the version in `package.json`.
-* Ensure the `CHANGELOG.md` is up to date.
+* Generate Release Notes using the provided script `./script/generate_release_notes.sh 1.2.3`
```
Comment (andrewjtait) [inline README.md]: dont need number on the end of this any more

File: script/generate-release-notes.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  | head -1 \
+  | awk -F: '{ print $2 }' \
+  | sed 's/[",]//g')
+
+dot_version=`echo $PACKAGE_VERSION | sed -E "s/-rc.*$//"`
+dash_version=`echo ${dot_version} | tr '.' '_'`
+
+RELEASE_NOTES_FILE="release_notes/v${dash_version}.html.md"
+
+renogen --format markdown v${dot_version} >> ${RELEASE_NOTES_FILE}
```
Comment (chrisbarber86) [inline script/generate-release-notes.sh]: We'll need a Gemfile/Gemfile.lock to ensure `renogen` is present

Comment (SeanArmstrong) [reply script/generate-release-notes.sh]: Do we as its installed globally on a machine. Or can be. Didn't want to cruft up the repo

Comment (SeanArmstrong) [reply script/generate-release-notes.sh]: Although if we want the check as part of build we will probably need it

Comment (chrisbarber86) [reply script/generate-release-notes.sh]: Global install is fine, we need instructions for that somewhere though right?

**Pull request merged**